### ISSUE TITLE
Assert failure at src/core/dbus-execute.c:2758 causes crash on KDE session start

### ISSUE THREAD

**Issue by tux3** [systemd/systemd#33015]
### systemd version the issue has been seen with

systemd_256~rc3-2

(Note that the mailing list encouraged me to report this on Github (https://lists.freedesktop.org/archives/systemd-devel/2024-May/050280.html), but the issue template claims only non-rc releases are in scope.  
Since Debian's libsystemd 255 does not crash on my system, this seems to be a new bug in the release candidate, but please do let me know if I should have waited for the final release of 256 to report this crash.)

### Used distribution

Debian sid

### Linux kernel version used

6.8.9-amd64

### CPU architectures issue was seen on

x86_64

### Component

systemd

### Expected behaviour you didn't see

KDE desktop starting normally.

### Unexpected behaviour you saw

Core dump in `systemd` following an assertion failure [at src/core/dbus-execute.c:2758](https://github.com/systemd/systemd/blob/main/src/core/dbus-execute.c#L2758) in `bus_exec_context_set_transient_property()`.

This results in a black screen after login, and stays stuck there until I manually restart SDDM from another tty.

### Steps to reproduce the problem

Unfortunately it is not easy for me to make a minimal reproducer, since the crash happens on login and results in a black screen. I doesn't happen if I start a KDE-in-KDE session, or if i start components (kwin, plasmashell, ..) manually.  
So while I can reproduce the crash 100% of the time, it's very much not a minimal repro...

What I do to reproduce is to [build KDE plasma from source](https://develop.kde.org/docs/getting-started/building/kdesrc-build-compile/) on a Debian sid, by running `kdesrc-build workspace`, then:
- Install the from-source KDE session: `./kde/build/plasma-workspace/login-sessions/install-sessions.sh`
- Reboot into the stock Debian KDE session (not built from source)
- Log out of my stock Debian KDE session, going back to SDDM
- Now select KDE (Development, Wayland) in SDDM (built from source)
- Click login, receive black screen, keep both pieces

### Additional program output to the terminal or log subsystem illustrating the issue

```sh
journalctl -r around crash:

mai 25 17:30:57 home systemd[1]: user@1000.service: Main process exited, code=dumped, status=6/ABRT
mai 25 17:30:57 home (sd-pam)[85099]: pam_unix(systemd-user:session): session closed for user tux3
mai 25 17:30:57 home systemd[1]: Started colord.service - Manage, Install and Generate Color Profiles.
mai 25 17:30:57 home dbus-daemon[84697]: [system] Successfully activated service 'org.freedesktop.ColorManager'
mai 25 17:30:57 home ksmserver[85480]: kf.kio.gui: Failed to launch process as service: "app-\\x2fusr\\x2fbin\\x2fkalendarac@736d4e661e99402f8f437351157fe5fc.service" "org.freedesktop.DBus.Error.NoReply" "Message recipient disconnected from message bus without replying"
mai 25 17:30:57 home ksmserver[85480]: kf.kio.gui: Failed to launch process as service: "app-\\x2fusr\\x2fbin\\x2fdolphin@0ce9778a49ea4f8dacaa0ddd46f384e1.service" "org.freedesktop.DBus.Error.NoReply" "Message recipient disconnected from message bus without replying"
mai 25 17:30:57 home ksmserver[85480]: kf.kio.gui: Failed to launch process as service: "app-\\x2fusr\\x2fbin\\x2fkonsole@4149ccd079e74f36b2aa3053cad471e7.service" "org.freedesktop.DBus.Error.NoReply" "Message recipient disconnected from message bus without replying"
mai 25 17:30:57 home ksmserver[85480]: kf.kio.gui: Failed to launch process as service: "app-\\x2fopt\\x2fthunderbird\\x2fthunderbird@bce1658b650a4536a8c727b8169324fe.service" "org.freedesktop.DBus.Error.NoReply" "Message recipient disconnected from message bus without >
mai 25 17:30:57 home ksmserver[85480]: kf.kio.gui: Failed to launch process as service: "app-\\x2fopt\\x2ffirefox\\x2ffirefox@3eff6298584d43a7be7f87753362d87c.service" "org.freedesktop.DBus.Error.NoReply" "Message recipient disconnected from message bus without replying"
mai 25 17:30:57 home systemd[1]: systemd-coredump@2-85716-0.service: Deactivated successfully.
mai 25 17:30:57 home systemd-coredump[85724]: [🡕] Process 85098 (systemd) of user 1000 dumped core.
                                              
                                              Module libzstd.so.1 from deb libzstd-1.5.5+dfsg2-2.amd64
                                              Module libblkid.so.1 from deb util-linux-2.40.1-1.amd64
                                              Module libmount.so.1 from deb util-linux-2.40.1-1.amd64
                                              Module libsystemd-shared-256.so from deb systemd-256~rc3-2.amd64
                                              Module libsystemd-core-256.so from deb systemd-256~rc3-2.amd64
                                              Module systemd from deb systemd-256~rc3-2.amd64
                                              Stack trace of thread 85098:
                                              #0  0x00007fcdd1ea6b1c __pthread_kill_implementation (libc.so.6 + 0x8bb1c)
                                              #1  0x00007fcdd1e584e2 __GI_raise (libc.so.6 + 0x3d4e2)
                                              #2  0x00007fcdd1e414ed __GI_abort (libc.so.6 + 0x264ed)
                                              #3  0x00007fcdd208f4f4 n/a (libsystemd-shared-256.so + 0x8f4f4)
                                              #4  0x00007fcdd2486c0a bus_exec_context_set_transient_property (libsystemd-core-256.so + 0x86c0a)
                                              #5  0x00007fcdd2495bdb bus_service_set_property (libsystemd-core-256.so + 0x95bdb)
                                              #6  0x00007fcdd24a1e5e bus_unit_set_properties (libsystemd-core-256.so + 0xa1e5e)
                                              #7  0x00007fcdd248dd2d n/a (libsystemd-core-256.so + 0x8dd2d)
                                              #8  0x00007fcdd248dfa2 n/a (libsystemd-core-256.so + 0x8dfa2)
                                              #9  0x00007fcdd22673f6 n/a (libsystemd-shared-256.so + 0x2673f6)
                                              #10 0x00007fcdd226873b bus_process_object (libsystemd-shared-256.so + 0x26873b)
                                              #11 0x00007fcdd227a936 n/a (libsystemd-shared-256.so + 0x27a936)
                                              #12 0x00007fcdd227ad6c n/a (libsystemd-shared-256.so + 0x27ad6c)
                                              #13 0x00007fcdd22d5c91 n/a (libsystemd-shared-256.so + 0x2d5c91)
                                              #14 0x00007fcdd22d5fad sd_event_dispatch (libsystemd-shared-256.so + 0x2d5fad)
                                              #15 0x00007fcdd22d78f8 sd_event_run (libsystemd-shared-256.so + 0x2d78f8)
                                              #16 0x00007fcdd24f1eed manager_loop (libsystemd-core-256.so + 0xf1eed)
                                              #17 0x000055b51772d973 n/a (systemd + 0x7973)
                                              #18 0x00007fcdd1e42c8a __libc_start_call_main (libc.so.6 + 0x27c8a)
                                              #19 0x00007fcdd1e42d45 __libc_start_main_impl (libc.so.6 + 0x27d45)
                                              #20 0x000055b5177301e1 n/a (systemd + 0xa1e1)
                                              ELF object binary architecture: AMD x86-64
mai 25 17:30:57 home systemd[1]: Starting colord.service - Manage, Install and Generate Color Profiles...
mai 25 17:30:57 home aa-notify[85747]: ERROR: Cannot read /var/log/kern.log. Please check permissions.
mai 25 17:30:57 home dbus-daemon[84697]: [system] Activating via systemd: service name='org.freedesktop.ColorManager' unit='colord.service' requested by ':1.23' (uid=1000 pid=85479 comm="/usr/bin/kded5")
mai 25 17:30:57 home smart-notifier[85660]: Failed to load module "xapp-gtk3-module"
mai 25 17:30:57 home kded5[85479]: Imported file "/usr/share/khotkeys/konqueror_gestures_kde321.khotkeys"
mai 25 17:30:57 home kded5[85479]: Imported file "/usr/share/khotkeys/kde32b1.khotkeys"
mai 25 17:30:57 home kded5[85479]: Imported file "/usr/share/khotkeys/defaults.khotkeys"
mai 25 17:30:57 home plasmashell[85566]: starting invalid corona "org.kde.plasma.desktop"
mai 25 17:30:57 home kded5[85479]: true
mai 25 17:30:57 home kded5[85479]: Version 2 File!
mai 25 17:30:57 home kded5[85479]: Reloading the khotkeys configuration
mai 25 17:30:57 home plasmashell[85566]: kf.windowsystem: Could not find any platform plugin
mai 25 17:30:57 home kded5[85479]: QDBusConnection: name 'org.freedesktop.UDisks2' had owner '' but we thought it was ':1.42'
mai 25 17:30:57 home kded5[85479]: Delayed initialization.
mai 25 17:30:57 home plasmashell[85566]: kf.package: Cannot set a path in a package without structure "org.kde.plasma.desktop"
mai 25 17:30:57 home plasmashell[85566]: kf.package: Invalid metadata for package structure "Plasma/Shell"
mai 25 17:30:57 home plasmashell[85566]: kf.package: Invalid metadata for package structure "Plasma/LookAndFeel"
mai 25 17:30:57 home systemd[1]: Started drkonqi-coredump-processor@2-85716-0.service - Pass systemd-coredump journal entries to relevant user for potential DrKonqi handling.
mai 25 17:30:57 home systemd[1]: Started systemd-coredump@2-85716-0.service - Process Core Dump (PID 85716/UID 0).
mai 25 17:30:57 home systemd-coredump[85716]: Process 85098 (systemd) of user 1000 terminated abnormally with signal 6/ABRT, processing...
mai 25 17:30:57 home (pcscd)[85720]: pcscd.service: Referenced but unset environment variable evaluates to an empty string: PCSCD_ARGS
mai 25 17:30:57 home kaccess[85689]: Xlib XKB extension major= 1  minor= 0
mai 25 17:30:57 home gdbus[85706]: Error: GDBus.Error:org.freedesktop.DBus.Error.ServiceUnknown: The name org.freedesktop.Tracker1 was not provided by any .service files
mai 25 17:30:57 home systemd[1]: Started pcscd.service - PC/SC Smart Card Daemon.
mai 25 17:30:57 home systemd[85098]: Assertion '__unique_prefix__expr_90' failed at src/core/dbus-execute.c:2758, function bus_exec_context_set_transient_property(). Aborting.
mai 25 17:30:57 home systemd[85098]: Started app-xdg\x2duser\x2ddirs@autostart.service - User folders update.
mai 25 17:30:57 home systemd[85098]: Started app-tracker\x2dstore@autostart.service - Tracker Store.
mai 25 17:30:57 home systemd[85098]: Condition check resulted in app-klipper@autostart.service - Klipper being skipped.
mai 25 17:30:57 home systemd[85098]: app-klipper@autostart.service: Skipped due to 'exec-condition'.
mai 25 17:30:57 home systemd[85098]: Finished plasma-ksplash-ready.service - KSplash "ready" Stage.
mai 25 17:30:57 home systemd[85098]: Started app-pkcs11\x2dregister@autostart.service - Install OpenSC PKCS#11 module.
```

**Comment by YHNdnzj** [comment]
Fix is waiting in #33016. But note that the assertion is triggered in an edge case, which would normally be refused. So I presume this is also a bug for KDE too.

**Comment by tux3** [comment]
Thank you very much.

I'm not a KDE developer, just a user, but if I understand correctly this looks like a KDE service does a call to systemd with an invalid WorkingDirectory argument, such that I'm not going to find any offending unit file on disk, but I would have to trace where the dbus call is coming from?  
If I can identify where the empty WorkingDirectory is coming from, I will definitely be happy to try and report this to KDE.
